### ISSUE TITLE
MonetDB server crashes in `HEAP_malloc`

### ISSUE THREAD

**Issue by fuboat** [MonetDB/MonetDB#7415]
**Describe the bug**
MonetDB server crashes in `HEAP_malloc`.

**To Reproduce**
```sql
 CREATE TABLE t1 AS SELECT     t1.value AS id,     CAST(         sys.group_concat(             CASE WHEN id.value IS NOT NULL THEN 'DB' ELSE '' END             , '')     AS BLOB) AS name FROM     sys.generate_series(0, 1000) AS t1 LEFT JOIN     sys.generate_series(0, 1000) AS id ON id.value < t1.value GROUP BY id ORDER BY id     ;
 insert into t1 (select * from t1)  ;
 insert into t1 (select * from t1)  ;
 insert into t1 (select * from t1)  ;
 insert into t1 (select * from t1)  ;
 DELETE FROM t1 WHERE id=1;  
 select nth_value(aa, 1) over (partition by bb) from analytics ;
 insert into t1 (select * from t1)  ;
 insert into t1 (select * from t1)  ; 
```

**Backtrace of the Crash**
```c
#0 0x7ffb416f2ce7 (HEAP_malloc+0xa7)
#1 0x7ffb4170319a (BLOBput+0x7a)
#2 0x7ffb4165effe (BATappend_or_update+0x15ee)
#3 0x7ffb416656fa (BATupdate+0x4a)
#4 0x7ffb405956c1 (append_col_execute+0x8b1)
#5 0x7ffb4058197b (append_col+0xcb)
#6 0x7ffb403e02b0 (mvc_append_wrap+0x660)
#7 0x7ffb4095829f (runMALsequence+0xf0f)
#8 0x7ffb4095d933 (DFLOWworker+0x4a3)
#9 0x7ffb417df8a7 (thread_starter+0xc7)
#10 0x7ffb402fa609 (start_thread+0xd9)
#11 0x7ffb4021d133 (clone+0x43)
```

**Software versions**
 - MonetDB version number: v11.47.11 (Jun2023-SP2)
 - OS and version: Ubuntu 20.04
 - From docker image monetdb/monetdb:Jun2023-SP2

An example of steps to reproduce in docker container (**write the poc into /tmp/test.sql first**):
```bash
docker container rm monetdb_test -f                                 # remove the old one
docker run -e MDB_DB_ADMIN_PASS=monetdb --name monetdb_test -itd monetdb/monetdb:Jun2023-SP2
                                                                    # start monetdb in docker
docker exec monetdb_test bash -c 'echo -n -e "user=monetdb\npassword=monetdb\n" > /root/.monetdb'
                                                                    # set username and password
sleep 10                                                            # wait the server start
docker exec monetdb_test bash -c "grep '/usr/bin/[m]server5' /proc/*/cmdline -lR"
                                                                    # check the existence of mserver5 process
cat /tmp/test.sql | docker exec -i monetdb_test mclient monetdb     # run test case
docker exec monetdb_test bash -c "grep '/usr/bin/[m]server5' /proc/*/cmdline -lR" || { echo "[CRASH] Mserver5 process does NOT exist."; exit 1; }
                                                                    # check the existence of mserver5 process
echo "[INFO] Server normal."
```

It can also be reproduced on MonetDB Dec2023_root, self-compiled by `CC=clang-12 CXX=clang++-12 cmake /root/monetdb -DCMAKE_BUILD_TYPE=RelWithDebInfo`.

**Issue labeling**
bug

**Comment by mvdvm** [comment]
Analysis: Problem is caused by the DELETE statement. Without the DELETE is runs fine.
Added a test for this bug.

Please note that the `select nth_value(aa, 1) over (partition by bb) from analytics ;` statement produces an error. as table  analytics is not defined. This statement is not needed to reproduce the crash.

**Comment by sjoerdmullender** [comment]
Bug is fixed in the Dec2023 branch.
